Fix issue with power balancer on new-profile branch

cmcantalupo opened the pull request:
Fix bug in power balancer on new-profile branch and clean up the tests a bit.

cmcantalupo committed:
Get rid of skip decorator for long tests

- When using __main__.py the long tests are still skipped unless
  the environment variable is set, but when executing the individual
  files, it is not regarded.
- If you are running a test using the individual file, then it doesn't
  make much sense to skip the test (obviously you meant to run it).
- This is technical debt from before the tests were split into separate
  files.

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Add more checks to SampleAggregator unit tests for the *last() mothods

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Clean up test_power_balancer test a bit and add some more report signals

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Fix power balancer on new-profile branch

- When the switch to using the SampleAggregator was made
  the call to update it each sample was omitted.
- Add the call to update the aggregators on each control loop.
- This fixes #1559 from github issues.

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

bgeltz commented on integration/test/__main__.py:
```diff
@@ -56,11 +52,16 @@
 from test_hint_time import *
 from test_progress import *
 
+if 'GEOPM_RUN_LONG_TESTS' in os.environ
```
missing ```:``` at EOL

Was this tested?

cmcantalupo replied:
Obviously not :( thanks, will run main...

bgeltz commented on integration/test/__main__.py:
```diff
@@ -56,11 +52,16 @@
+    from test_power_balancer import *
+    from test_power_governor import *
+    from test_scaling_region import *
+    from test_timed_scaling_region import *
```
```
else:
    print a nice warning message about skipped tests
```

bgeltz commented on test/SampleAggregatorTest.cpp:
```diff
@@ -339,9 +339,29 @@ TEST_F(SampleAggregatorTest, epoch_application_total)
 
     EXPECT_DOUBLE_EQ(2.0, m_agg->sample_region(M_SIGNAL_TIME, reg_normal));
     EXPECT_DOUBLE_EQ(2.0, m_agg->sample_region(M_SIGNAL_TIME, GEOPM_REGION_HASH_UNMARKED));
-    // First epoch observed at step == 2, app finished at step == 4.  4 - 2 = 2
```
Some comments in the new code would really help in understanding what your changes are doing.  It took us a while to decipher this thing last night.

cmcantalupo committed:
Fix integration test __main__.py and add skip warnings

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Clean up SampleAggregatorTest

- More comments, and switched WillRepeatedly() -> WillOnce() calls where
  appropriate.

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

bgeltz commented on test/SampleAggregatorTest.cpp:
```diff
@@ -298,50 +298,83 @@ TEST_F(SampleAggregatorTest, epoch_application_total)
-    // First epoch observed at step == 2, app finished at step == 4.  4 - 2 = 2
-    EXPECT_DOUBLE_EQ(2.0, m_agg->sample_epoch(M_SIGNAL_TIME));
+
+    /// Run through the same three region hashes with the epoch set to two
```
```suggestion
    // Run through the same three region hashes with the epoch set to two
```

bgeltz committed:
Update test/SampleAggregatorTest.cpp
